### PR TITLE
feat(testrunner): support --test-match option

### PR DESCRIPTION
**Pull request by JoelEinbinder**
also adds support for --test-ignore.

--test-match makes it the `.spec.ts` format for tests configurable. --test-ignore allows for ignoring some paths, which is important for node_modules. micromatch is used to parse the patterns.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "check-deps": "node utils/check_deps.js",
     "build-driver": "pkg --public --targets node12-linux-x64,node12-macos-x64,node12-win-x64 --out-path=drivers packages/playwright-driver/main.js",
     "build-testrunner": "tsc -p test-runner",
-    "test-testrunner": "node test-runner/cli test-runner/test"
+    "test-testrunner": "node test-runner/cli --test-ignore='**/assets/**' test-runner/test"
```
Comment (pavelfeldman) [inline package.json:35]: `--ignore` ?

File: package.json [hunk 2]
```diff
@@ -79,6 +80,7 @@
     "esprima": "^4.0.0",
     "expect": "^26.3.0",
     "formidable": "^1.2.1",
+    "micromatch": "^4.0.2",
```
Comment (pavelfeldman) [inline package.json:83]: We try to not pull non-dev dependencies into the system. Check out `globToRegex` in Playwright, should we reuse/copy it instead?

File: test-runner/src/cli.ts [hunk 3]
```diff
@@ -40,6 +41,8 @@ program
   .option('--output <outputDir>', 'Folder for output artifacts, default: test-results', path.join(process.cwd(), 'test-results'))
   .option('--timeout <timeout>', 'Specify test timeout threshold (in milliseconds), default: 10000', '10000')
   .option('-u, --update-snapshots', 'Use this flag to re-record every snapshot that fails during this test run')
+  .option('--test-match <pattern>', 'Pattern used to find test files', '**/?(*.)+(spec|test).[jt]s')
```
Comment (pavelfeldman) [inline test-runner/src/cli.ts:44]: How does this relate to `command.args.slice(1)` ? I.e. when do I use one vs the other?

File: package.json (hunk 2 above)
Comment (JoelEinbinder) [reply package.json:83]: Minimatch/micromatch implement bash glob patterns. Minimatch (20M weekly downloads) is used by npm itself, mocha, eslint and Micromatch (24M weekly downloads) is a smaller competitor used by Jest, webpack, and others. This is the defacto standard glob syntax for the node world. `globToRegex` is missing standard features like '!' negation. I also do not trust it to work with files, and don't think it is worth any time.

Micromatch is already included in our dependency tree via `expect`.

File: test-runner/src/cli.ts (hunk 3 above)
Comment (JoelEinbinder) [reply test-runner/src/cli.ts:44]: Aside from the difference in parsing the patterns, test match is ANDed with filters. See https://jestjs.io/docs/en/configuration#testmatch-arraystring and https://jestjs.io/docs/en/cli#jest-regexfortestfiles.

Comment (JoelEinbinder) [reply test-runner/src/cli.ts:44]: You use a filter "mouse" when you want to run "mouse.spec.ts" not "keyboard.spec.ts". You use --test-match when you want to run "mouse.spec.ts" not "mouse.ts".

File: package.json (hunk 1 above)
Comment (JoelEinbinder) [reply package.json]: done

File: package.json (hunk 2 above)
Comment (pavelfeldman) [reply package.json:83]: Okay

File: test-runner/src/cli.ts (hunk 3 above)
Comment (pavelfeldman) [reply test-runner/src/cli.ts:44]: The second part seems confusing, why can't I just say `npm run test mouse.ts`?

**Review by pavelfeldman — APPROVED** [review]

**Comment by JoelEinbinder** [comment]
Will re-send this in the new repo.

**Pull request closed without merging**